src/main/java/com/hotel/mappers/book/request/GuestMapper.java

```diff
@@ -0,0 +1,53 @@
+package com.hotel.mappers.book.request;
+
+import com.hotel.service.common.Address;
+import com.hotel.service.common.UserInfo;
+import org.opentravel.ota._2003._05.*;
+import org.springframework.stereotype.Component;
+
+@Component
+public class GuestMapper {
+
+    public ArrayOfResGuestsTypeResGuest map(UserInfo userInfo) {
+        ArrayOfResGuestsTypeResGuest arrayOfResGuestsTypeResGuest = new ArrayOfResGuestsTypeResGuest();
+        ArrayOfResGuestsTypeResGuest.ResGuest resGuest = new ArrayOfResGuestsTypeResGuest.ResGuest();
+        ArrayOfProfilesTypeProfileInfo profiles = new ArrayOfProfilesTypeProfileInfo()  ;
+        resGuest.setProfiles(profiles);
+        ArrayOfProfilesTypeProfileInfo.ProfileInfo profileInfo = new ArrayOfProfilesTypeProfileInfo.ProfileInfo();
+        ProfileType profile = new ProfileType();
+        CustomerType customer = new CustomerType();
+
+        PersonNameType personName = new PersonNameType();
+        personName.getNamePrefix().add(userInfo.getNamePrefix());
+        personName.getGivenName().add(userInfo.getFirstName());
+        personName.setSurname(userInfo.getLastName());
+        customer.setPersonName(personName);
+
+        CustomerType.Telephone telephone = new CustomerType.Telephone();
+        telephone.setPhoneNumber(userInfo.getPhoneNumber());
+        customer.getTelephone().add(telephone);
+        EmailType emailType = new EmailType();
+        emailType.setValue(userInfo.getEmail());
+        customer.getEmail().add(emailType);
+        customer.getAddress().add(mapAddress(userInfo.getAddress()));
+
```

Comment:
Can we move 

        PersonNameType personName = new PersonNameType();
        personName.getNamePrefix().add(userInfo.getNamePrefix());
        personName.getGivenName().add(userInfo.getFirstName());
        personName.setSurname(userInfo.getLastName());
        customer.setPersonName(personName);

        CustomerType.Telephone telephone = new CustomerType.Telephone();
        telephone.setPhoneNumber(userInfo.getPhoneNumber());
        customer.getTelephone().add(telephone);
        EmailType emailType = new EmailType();
        emailType.setValue(userInfo.getEmail());
        customer.getEmail().add(emailType);
        customer.getAddress().add(mapAddress(userInfo.getAddress()));

to other mapper and return Customer from there and also address can be another mapper

Reply:
will do later